index.js

```diff
@@ -805,159 +811,185 @@ function buildValue (location, input) {
               ${nestedResult}
           `
-        }
+          }
 
-        code += `
+          code += `
           else throw new Error(\`The value $\{JSON.stringify(${input})} does not match schema definition.\`)
         `
-      } else if (isEmpty(schema)) {
-        code += `
+        } else if (isEmpty(schema)) {
+          code += `
           json += JSON.stringify(${input})
         `
-      } else if ('const' in schema) {
-        code += `
+        } else if ('const' in schema) {
+          code += `
           if(ajv.validate(${JSON.stringify(schema)}, ${input}))
             json += '${JSON.stringify(schema.const)}'
           else
             throw new Error(\`Item $\{JSON.stringify(${input})} does not match schema definition.\`)
         `
-      } else if (schema.type === undefined) {
-        code += `
+        } else if (schema.type === undefined) {
+          code += `
           json += JSON.stringify(${input})
         `
-      } else {
-        throw new Error(`${schema.type} unsupported`)
-      }
-      break
-    default:
-      if (Array.isArray(type)) {
-        let sortedTypes = type
-        const nullable = schema.nullable === true || type.includes('null')
+        } else {
+          throw new Error(`${schema.type} unsupported`)
+        }
+        break
 
-        if (nullable) {
-          sortedTypes = sortedTypes.filter(type => type !== 'null')
-          code += `
+      default:
+        if (Array.isArray(type)) {
+          let sortedTypes = type
+          const nullable = schema.nullable === true || type.includes('null')
+
+          if (nullable) {
+            sortedTypes = sortedTypes.filter(type => type !== 'null')
+            code += `
             if (${input} === null) {
               json += null
             } else {`
-        }
+          }
 
-        const locationClone = clone(location)
-        sortedTypes.forEach((type, index) => {
-          const statement = index === 0 ? 'if' : 'else if'
-          locationClone.schema.type = type
-          const nestedResult = buildValue(locationClone, input)
-          switch (type) {
-            case 'string': {
-              code += `
-                ${statement}(${input} === null || typeof ${input} === "${type}" || ${input} instanceof RegExp || (typeof ${input} === "object" && Object.prototype.hasOwnProperty.call(${input}, "toString")))
+          const locationClone = clone(location)
+          sortedTypes.forEach((type, index) => {
+            const statement = index === 0 ? 'if' : 'else if'
+            locationClone.schema.type = type
+            const nestedResult = buildValue(locationClone, input)
+            switch (type) {
+              case 'string': {
+                code += `
+                ${statement}(${input} === null || typeof ${input} === "${type}" || ${input} instanceof RegExp || (typeof ${input} === "object" && Object.hasOwnProperty.call(${input}, "toString")))
```

Comment:
Please rebase your branch to the master before merging it. Otherwise, you will overwrite #510.